src/simcore_service_deployment_agent/portainer.py

```diff
@@ -174,20 +176,48 @@ async def update_stack(
         endpoint_id = await get_first_endpoint_id(base_url, app_session, bearer_code)
         log.debug("Determined the following endpoint id: %i", endpoint_id)
     headers = {"Authorization": f"Bearer {bearer_code}"}
-    body_data = {"StackFileContent": json.dumps(stack_cfg, indent=2)}
+    body_data = {"StackFileContent": json.dumps(stack_cfg, indent=2), "pullImage": True}
     log.debug("StackFileContent:")
     log.debug(json.dumps(stack_cfg, indent=2, sort_keys=True))
     url = (
         URL(base_url)
         .with_path(f"api/stacks/{stack_id}")
         .with_query({"endpointId": endpoint_id, "method": "string", "type": 1})
     )
-    time.sleep(0.5)
     try:
         data = await _portainer_request(
             url, app_session, "PUT", headers=headers, json=body_data
         )
         log.debug("updated stack: %s", data)
-    except asyncio.exceptions.TimeoutError as err:
-        print("ERROR")
-        print(f"{err}")
+    except asyncio.exceptions.TimeoutError:
+        log.exception("Unhandled error while deleting stack %s", url)
+
+
+async def delete_stack(
+    base_url: URL,
+    app_session: ClientSession,
+    bearer_code: str,
+    stack_id: int,
+    endpoint_id: int,
+):  # pylint: disable=too-many-arguments
```

Comment:
it might be worth setting pylint number of arguments to something bigger than 5.
```suggestion
) -> None:  # pylint: disable=too-many-arguments
```

Reply:
it is set to 10, will remove the pylint directive